### ISSUE TITLE
Rule that checks if you remembered to call super.method()

### ISSUE THREAD

**Issue by trusktr** [eslint/eslint#11969]
**Please describe what the rule should do:**

Tells you if you forgot to call `super.method()` in a subclass. This is similar to the existing `call-super` rule, but `call-super` currently only applies to constructors.

Perhaps instead of making a new rule, `call-super` could be extended to also include methods.

More often than not, I want to call a super method in a subclass method.

**What category of rule is this? (place an "X" next to just one item)**

[x] Warns about a potential error (problem)
[ ] Suggests an alternate way of doing something (suggestion)
[ ] Enforces code style (layout)
[ ] Other (please specify:)

**Provide 2-3 code examples that this rule will warn about:**

```js
class Foo {
 method() {...}
}

class Bar extends Foo {
  method() {
    console.log(this.something)
    // oops, forgot to call super.method()
  }
}
```

**Why should this rule be included in ESLint (instead of a plugin)?**

Because, easier for people to turn it on.

**Are you willing to submit a pull request to implement this rule?**

I might be, time-allowing, but I haven't done one before.

**Comment by platinumazure** [comment]
I'm in favor of this if we can reliably know what methods are defined on the superclass. I'm not sure we can know that without type knowledge.

The `constructor` case is different, because we know all classes have a constructor.

**Comment by trusktr** [comment]
Ah true. I took it for granted, without thinking, that the base class would be known (accustomed to TypeScript).

Maybe this is a plugin more suited for the TypeScript ESLint plugins. Though it would be nice to have it for JS too if it were possible.

**Comment by trusktr** [comment]
Is there currently any cross-file tracking in ESLint? F.e., if a class is imported from one file, and used somewhere in the `extends` expression of another file, maybe the static analysis could be possible in this case.

**Comment by platinumazure** [comment]
@trusktr No, currently there is no cross-file tracking in ESLint core.
